packages/api/src/command/hie/unlink-patient-from-organization.ts

```diff
@@ -0,0 +1,345 @@
+import { Config } from "../../shared/config";
+import { makeFhirApi } from "../../external/fhir/api/api-factory";
+
+const s3Utils = new S3Utils(Config.getAWSRegion());
+const s3ConversionResultBucketName = getEnvVarOrFail("CONVERSION_RESULT_BUCKET_NAME");
+const s3MedicalDocumentsBucketName = getEnvVarOrFail("MEDICAL_DOCUMENTS_BUCKET_NAME");
+
+type UnlinkPatientFromOrganizationParams = {
+  cxId: string;
+  patientId: string;
+  oid: string;
+  dryRun?: boolean;
+};
+
+function getDryRunPrefix(dryRun?: boolean) {
+  return dryRun ? "[DRY RUN] " : "";
+}
+
+export async function unlinkPatientFromOrganization({
+  cxId,
+  patientId,
+  oid,
+  dryRun = false,
+}: UnlinkPatientFromOrganizationParams): Promise<void> {
+  const dryRunMsg = getDryRunPrefix(dryRun);
+  const { log } = out(
+    `${dryRunMsg}unlinkPatientFromOrganization - patient ${patientId} - cxId ${cxId}`
+  );
+  log(`Unlinking patient from organization ${oid}`);
+
+  const documents = await getDocuments({ cxId, patientId });
+
+  if (documents.length === 0) {
+    log(`No documents found for patient ${patientId}`);
+    return;
+  }
+
+  const documentsWithOid = getDocumentsWithOid(documents, oid);
+
+  if (documentsWithOid.length === 0) {
+    log(`No documents found for patient ${patientId} with oid ${oid}`);
+    return;
+  }
+
+  log(`Found ${documentsWithOid.length} documents to process`);
+
+  const [cqPatientData, cwPatientData] = await Promise.all([
+    getCQPatientData({ id: patientId, cxId }),
+    getCwPatientData({ id: patientId, cxId }),
+  ]);
+
+  const cwLink = findCwLinkWithOid(cwPatientData?.data, oid);
+  const cqLink = findCqLinkWithOid(cqPatientData?.data, oid);
+
+  await findAndInvalidateLinks(cwLink, cqLink, cxId, patientId, dryRun, log);
+
+  const errors: { documentId: string; error: unknown }[] = [];
+
+  for (const document of documentsWithOid) {
+    const fileName = getS3FileNameFromDocument(document, log);
+    if (!fileName) {
+      log(`Skipping document ${document.id} - no filename found`);
+      continue;
+    }
+
+    try {
+      log(`Processing document ${document.id}`);
+      await Promise.all([
+        findAndRemoveConversionResultsFromS3(fileName, dryRun, log),
+        findAndRemoveMedicalDocumentFromS3(fileName, dryRun, log),
+        findAndRemoveConsolidatedDocumentFromS3(cxId, patientId, dryRun, log),
+        deleteFromOpenSearch(document.id, dryRun, log),
+      ]);
+
+      // Only delete FHIR resource if all other operations succeeded
+      await deleteFhirResource(cxId, document.id, dryRun, log);
+      log(`Successfully processed document ${document.id}`);
+    } catch (error) {
+      log(`Failed to process document ${document.id}: ${errorToString(error)}`);
+      errors.push({ documentId: document.id, error });
+      continue;
+    }
+  }
+
+  if (errors.length > 0) {
+    capture.error("Failed to process some documents during unlink", { extra: { errors } });
```

Comment:
can we add other params here before `error`, please?

 cxId,
  patientId,
  oid